stl/inc/span

```diff
@@ -0,0 +1,727 @@
+        : _Mybase(static_cast<size_type>(_RANGES size(_Range))), _Mydata(_RANGES data(_Range)) {}
+
+    template <class _OtherTy, size_t _OtherExtent>
+        requires (_Extent == dynamic_extent || _Extent == _OtherExtent)
+              && is_convertible_v<_OtherTy (*)[], element_type (*)[]>
+    constexpr span(const span<_OtherTy, _OtherExtent>& _Other) noexcept
+        : _Mybase(_Other.size()), _Mydata(_Other.data()) {}
+    // clang-format on
+
+#else // ^^^ __cpp_lib_concepts / !__cpp_lib_concepts vvv
+
+    template <size_t _Ext = _Extent, enable_if_t<_Ext == 0 || _Ext == dynamic_extent, int> = 0>
+    constexpr span() noexcept {}
+
+    constexpr span(pointer _Ptr, size_type _Count) noexcept // strengthened
+        : _Mybase(_Count), _Mydata(_Ptr) {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        if constexpr (_Extent != dynamic_extent) {
+            _STL_VERIFY(_Count == _Extent, "span construction from invalid range as count != extent");
+        }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+    }
+
+    constexpr span(pointer _First, pointer _Last) noexcept // strengthened
+        : _Mybase(static_cast<size_type>(_Last - _First)), _Mydata(_First) {
+        _Adl_verify_range(_First, _Last);
+#if _CONTAINER_DEBUG_LEVEL > 0
+        if constexpr (_Extent != dynamic_extent) {
+            _STL_VERIFY(_Last - _First == _Extent, "span construction from invalid range as last - first != extent");
+        }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+    }
+
+    template <size_t _Size, enable_if_t<_Extent == dynamic_extent || _Extent == _Size, int> = 0>
+    constexpr span(element_type (&_Arr)[_Size]) noexcept : _Mybase(_Size), _Mydata(_Arr) {}
+
+    template <size_t _Size, enable_if_t<_Extent == dynamic_extent || _Extent == _Size, int> = 0>
+    constexpr span(array<value_type, _Size>& _Arr) noexcept : _Mybase(_Size), _Mydata(_Arr.data()) {}
+
+    template <size_t _Size, size_t _Ext = _Extent,
+        enable_if_t<conjunction_v<bool_constant<_Ext == dynamic_extent || _Ext == _Size>,
+                        is_convertible<const value_type*, element_type*>>,
+            int> = 0>
+    constexpr span(const array<value_type, _Size>& _Arr) noexcept : _Mybase(_Size), _Mydata(_Arr.data()) {}
+
+    template <class _Rng, enable_if_t<conjunction_v<bool_constant<_Extent == dynamic_extent>,
+                                          _Is_span_compatible_range<_Rng, element_type>>,
+                              int> = 0>
+    constexpr span(_Rng& _Range) : _Mybase(static_cast<size_type>(_STD size(_Range))), _Mydata(_STD data(_Range)) {}
+
+    template <class _Rng, enable_if_t<conjunction_v<bool_constant<_Extent == dynamic_extent>,
+                                          _Is_span_compatible_range<const _Rng, element_type>>,
+                              int> = 0>
+    constexpr span(const _Rng& _Range)
+        : _Mybase(static_cast<size_type>(_STD size(_Range))), _Mydata(_STD data(_Range)) {}
+
+    template <class _OtherTy, size_t _OtherExtent,

```

Comment:
I think this is the culprit for the unused `_SpanExtent` functionality. Originally this would call `_Mybase(_OtherExtent)`. rather than `_Mybase(_Other.size())`.

Interestingly this would only matter if dynamic->static construction would be allowed as we would then have to verify `_Extent == _OtherExtent.size()`. However, as that conversion is currently not allowed by the standard we are save.